### ISSUE TITLE
CloudWatchClient.getMetricData doesn't serialize start and end time correctly

### ISSUE THREAD

**Issue by rohan-panchal** [awslabs/aws-sdk-swift#978]
### Describe the bug

CloudWatchClient.getMetricData returns a validation error, related to the start and end time being nil, while they are not.

### Expected Behavior

Passing Foundation.Date objects (which is the type specified in `GetMetricsInput`) for `startTime` and `endTime` should serialize correctly and not run into nil value validation errors.

### Current Behavior

Request Body:

```
RequestBody: Action=GetMetricData&MetricDataQueries.member.1.Id=DiskWriteOps&MetricDataQueries.member.1.MetricStat.Metric.MetricName=DiskWriteOps&MetricDataQueries.member.1.MetricStat.Period=300&MetricDataQueries.member.1.MetricStat.Stat=DiskWriteOps&MetricDataQueries.member.1.Period=5&MetricDataQueries.member.10.Id=StatusCheckFailed&MetricDataQueries.member.10.MetricStat.Metric.MetricName=StatusCheckFailed&MetricDataQueries.member.10.MetricStat.Period=300&MetricDataQueries.member.10.MetricStat.Stat=StatusCheckFailed&MetricDataQueries.member.10.Period=5&MetricDataQueries.member.11.Id=CPUSurplusCreditBalance&MetricDataQueries.member.11.MetricStat.Metric.MetricName=CPUSurplusCreditBalance&MetricDataQueries.member.11.MetricStat.Period=300&MetricDataQueries.member.11.MetricStat.Stat=CPUSurplusCreditBalance&MetricDataQueries.member.11.Period=5&MetricDataQueries.member.12.Id=NetworkPacketsOut&MetricDataQueries.member.12.MetricStat.Metric.MetricName=NetworkPacketsOut&MetricDataQueries.member.12.MetricStat.Period=300&MetricDataQueries.member.12.MetricStat.Stat=NetworkPacketsOut&MetricDataQueries.member.12.Period=5&MetricDataQueries.member.13.Id=StatusCheckFailed_System&MetricDataQueries.member.13.MetricStat.Metric.MetricName=StatusCheckFailed_System&MetricDataQueries.member.13.MetricStat.Period=300&MetricDataQueries.member.13.MetricStat.Stat=StatusCheckFailed_System&MetricDataQueries.member.13.Period=5&MetricDataQueries.member.14.Id=StatusCheckFailed_Instance&MetricDataQueries.member.14.MetricStat.Metric.MetricName=StatusCheckFailed_Instance&MetricDataQueries.member.14.MetricStat.Period=300&MetricDataQueries.member.14.MetricStat.Stat=StatusCheckFailed_Instance&MetricDataQueries.member.14.Period=5&MetricDataQueries.member.15.Id=CPUUtilization&MetricDataQueries.member.15.MetricStat.Metric.MetricName=CPUUtilization&MetricDataQueries.member.15.MetricStat.Period=300&MetricDataQueries.member.15.MetricStat.Stat=CPUUtilization&MetricDataQueries.member.15.Period=5&MetricDataQueries.member.16.Id=NetworkOut&MetricDataQueries.member.16.MetricStat.Metric.MetricName=NetworkOut&MetricDataQueries.member.16.MetricStat.Period=300&MetricDataQueries.member.16.MetricStat.Stat=NetworkOut&MetricDataQueries.member.16.Period=5&MetricDataQueries.member.17.Id=NetworkIn&MetricDataQueries.member.17.MetricStat.Metric.MetricName=NetworkIn&MetricDataQueries.member.17.MetricStat.Period=300&MetricDataQueries.member.17.MetricStat.Stat=NetworkIn&MetricDataQueries.member.17.Period=5&MetricDataQueries.member.2.Id=DiskWriteBytes&MetricDataQueries.member.2.MetricStat.Metric.MetricName=DiskWriteBytes&MetricDataQueries.member.2.MetricStat.Period=300&MetricDataQueries.member.2.MetricStat.Stat=DiskWriteBytes&MetricDataQueries.member.2.Period=5&MetricDataQueries.member.3.Id=DiskReadBytes&MetricDataQueries.member.3.MetricStat.Metric.MetricName=DiskReadBytes&MetricDataQueries.member.3.MetricStat.Period=300&MetricDataQueries.member.3.MetricStat.Stat=DiskReadBytes&MetricDataQueries.member.3.Period=5&MetricDataQueries.member.4.Id=MetadataNoToken&MetricDataQueries.member.4.MetricStat.Metric.MetricName=MetadataNoToken&MetricDataQueries.member.4.MetricStat.Period=300&MetricDataQueries.member.4.MetricStat.Stat=MetadataNoToken&MetricDataQueries.member.4.Period=5&MetricDataQueries.member.5.Id=DiskReadOps&MetricDataQueries.member.5.MetricStat.Metric.MetricName=DiskReadOps&MetricDataQueries.member.5.MetricStat.Period=300&MetricDataQueries.member.5.MetricStat.Stat=DiskReadOps&MetricDataQueries.member.5.Period=5&MetricDataQueries.member.6.Id=CPUCreditBalance&MetricDataQueries.member.6.MetricStat.Metric.MetricName=CPUCreditBalance&MetricDataQueries.member.6.MetricStat.Period=300&MetricDataQueries.member.6.MetricStat.Stat=CPUCreditBalance&MetricDataQueries.member.6.Period=5&MetricDataQueries.member.7.Id=CPUSurplusCreditsCharged&MetricDataQueries.member.7.MetricStat.Metric.MetricName=CPUSurplusCreditsCharged&MetricDataQueries.member.7.MetricStat.Period=300&MetricDataQueries.member.7.MetricStat.Stat=CPUSurplusCreditsCharged&MetricDataQueries.member.7.Period=5&MetricDataQueries.member.8.Id=NetworkPacketsIn&MetricDataQueries.member.8.MetricStat.Metric.MetricName=NetworkPacketsIn&MetricDataQueries.member.8.MetricStat.Period=300&MetricDataQueries.member.8.MetricStat.Stat=NetworkPacketsIn&MetricDataQueries.member.8.Period=5&MetricDataQueries.member.9.Id=CPUCreditUsage&MetricDataQueries.member.9.MetricStat.Metric.MetricName=CPUCreditUsage&MetricDataQueries.member.9.MetricStat.Period=300&MetricDataQueries.member.9.MetricStat.Stat=CPUCreditUsage&MetricDataQueries.member.9.Period=5&Version=2010-08-01&endTime=2023-05-01T19%3A04%3A40.933Z&startTime=2023-05-01T18%3A04%3A40.933Z
```

Response logs:

```
ResponseBody: <ErrorResponse xmlns="http://monitoring.amazonaws.com/doc/2010-08-01/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>2 validation errors detected: Value null at 'startTime' failed to satisfy constraint: Member must not be null; Value null at 'endTime' failed to satisfy constraint: Member must not be null</Message>
  </Error>
  <RequestId>7df685b0-7b9e-4d0f-a98d-458ce226c958</RequestId>
</ErrorResponse>
```

### Reproduction Steps

Code sample below: Just place credentials.

```
        let metricNames: [String] = [
            "DiskWriteOps",
            "DiskWriteBytes",
            "DiskReadBytes",
            "MetadataNoToken",
            "DiskReadOps",
            "CPUCreditBalance",
            "CPUSurplusCreditsCharged",
            "NetworkPacketsIn",
            "CPUCreditUsage",
            "StatusCheckFailed",
            "CPUSurplusCreditBalance",
            "NetworkPacketsOut",
            "StatusCheckFailed_System",
            "StatusCheckFailed_Instance",
            "CPUUtilization",
            "NetworkOut",
            "NetworkIn",
        ]
        let credentials = AWSCredentials(accessKey: "", secret: "")
        let credentialProvider = try AWSClientRuntime.AWSCredentialsProvider.fromStatic(credentials)
        let cloudWatchConfig = try await CloudWatchClient.CloudWatchClientConfiguration(credentialsProvider: credentialProvider,
                                                                                        region: region.name)
        
        cloudWatchConfig.clientLogMode = .requestAndResponseWithBody
        
        
        let cloudwatchClient = CloudWatchClient(config: cloudWatchConfig)
        
        let queries = metricNames.map {
            CloudWatchClientTypes.MetricStat(metric: .init(dimensions: nil, metricName: $0, namespace: nil), period: 300, stat: $0, unit: nil)
        }.map {
            CloudWatchClientTypes.MetricDataQuery(accountId: nil,
                                                  expression: nil,
                                                  id: $0.metric?.metricName,
                                                  label: nil,
                                                  metricStat: $0,
                                                  period: 5,
                                                  returnData: nil)
        }
        let endTime = Date.now
        let startTime = Date.init(timeIntervalSinceNow: -3600)
        debugPrint(endTime)
        debugPrint(startTime)
        
        let getMetricsInput = GetMetricDataInput(endTime: endTime,
                                                 maxDatapoints: nil,
                                                 metricDataQueries: queries,
                                                 nextToken: nil,
                                                 scanBy: nil,
                                                 startTime: startTime
        )
        
        let output = try await cloudwatchClient.getMetricData(input: getMetricsInput) // < = Error here
```

### Possible Solution

Looking at the request body, i wonder if the FormURLEncoder used in the `SerializableBodyMiddleware` doesn't serialize dates correctly?

### AWS SWIFT SDK version used

0.90 at first and then tested on 0.15.0

### Compiler and Version used

Xcode 14.2

### Operating System and version

iOS 16.2

**Comment by jbelkins** [comment]
Try to reproduce, observe actual format of datetimes

**Comment by jbelkins** [comment]
Duplicate of #1121 

Fixed in AWS SDK for Swift 0.26.0: https://github.com/awslabs/aws-sdk-swift/releases/tag/0.26.0

Closing this issue as Done.
